### ISSUE TITLE
HaveIBeenPwned needs API key

### ISSUE THREAD

**Issue by bowditch-c** [chrispetrou/WWWE#4]
The check for Have I been Pwned, no longer works as it requires an API key and subscription these days. So this will not function, and as a result the code as it stands will erroneously report that the email has been pwned when it hasn't.

**Comment by chrispetrou** [comment]
Yes you are right, I just checked it. [It was expected to happen](https://www.troyhunt.com/authentication-and-the-have-i-been-pwned-api/) since the v2 of the HIBP API would be replaced by the version 3. The reason I hadn't changed that until now was due to the fact that an v3 API key makes HIBP service a paid option. But since v2 is no more supported I have no option but to add an HIBP-API key as an additional option.

WWWE is an open-source project and I want it to be free of charge to use by anyone. So a solution that comes into mind right now is to check if a HIBP API key is provided and if it is then to use the service.

Thanks for taking the time to point that out I'll fix it asap. Also with the first chance I'll publish a newer version of the project that introduces some significant changes like:
- a database support
- flask integration
- json format of the results
